src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -193,20 +206,34 @@ export async function $onEmit(
                 let command = `dotnet ${resolvePath(options.csharpGeneratorPath)} --project-path ${outputFolder} ${newProjectOption}`;
                 console.info(command);
 
-                exec(command, (error, stdout, stderr) => {
-                    if (error) {
-                        console.log(`error: ${error.message}`);
-                    }
-                    else if (stderr) {
-                        console.log(`stderr: ${stderr}`);
-                    }
-                    console.log(`stdout: ${stdout}`);
-                });
-            }            
+                try {
+                    execSync(command, {stdio: 'inherit'});
+                } catch(error: any) {
+                    if (error.message) console.log(error.message);
+                    if (error.stderr) console.error(error.stderr);
+                    if (error.stdout) console.log(error.stdout);
+                }
+            }
+            
+            if (!options["save-inputs"]) {
+                // delete
+                deleteFile(resolvePath(generatedFolder, "cadl.json"));
```

Comment:
As those are intermediate files instead of input files,  shall we use 'save-intermediate' instead?

Reply:
The `input` is to say it is the input of the auotrest.csharp, `intermediate` is to say it is intermediate of the whole process. we will consider it related to autorest.csharp, so it is input files.

Reply:
@lirenhe `save-inputs` is current configuration flag in swagger flow/pipeline to save intermediate files: Configuration.json and codemodel.yml. I think we will keep consistent at this moment. and Michael is working on all those configurations, we will change this if needed after he finalizes all those.